execute/plugin.go

```diff
@@ -72,6 +80,21 @@ func NewPlugin(
 		tokenDataReader: tokenDataReader,
 		lggr:            lggr,
 	}
+
+	bgSyncCtx, bgSyncCf := context.WithCancel(context.Background())
+	p.bgSyncCancelFunc = bgSyncCf
+	p.bgSyncWG = &sync.WaitGroup{}
+	p.bgSyncWG.Add(1)
+	plugincommon.BackgroundReaderSync(
+		bgSyncCtx,
+		p.bgSyncWG,
+		lggr,
+		ccipReader,
+		syncTimeout(cfg.SyncTimeout),
+		time.NewTicker(syncFrequency(p.cfg.SyncFrequency)).C,
+	)
```

Comment:
idea: maybe this background reader syncer could be an object instead and the bg sync cancel / waitgroup could be implementation details instead? It'd have a `Close` method and a `Start` method similar to the `ServiceCtx`?

Reply:
done